src/fault/fiber/scheduler.cpp

```diff
@@ -0,0 +1,132 @@
+#include "fault/util.hpp"
+
+#include <yaclib/fault/detail/fiber/scheduler.hpp>
+
+#include <iostream>
+
+namespace yaclib::detail::fiber {
+
+Scheduler* current_scheduler = nullptr;
+
+static thread_local Fiber* current;
+
+Fiber* Scheduler::GetNext() {
+  YACLIB_DEBUG(_queue.empty(), "Queue can't be empty");
+  auto* next = PollRandomElementFromList(_queue);
+  return next;
+}
+
+bool Scheduler::IsRunning() const {
+  return _running;
+}
+
+void Scheduler::Suspend() {
+  YACLIB_DEBUG(current == nullptr, "Current can't be null");
+  auto* fiber = current;
+  fiber->Yield();
+}
+
+void Scheduler::Stop() {
+  Scheduler::Set(nullptr);
+  YACLIB_DEBUG(_running, "scheduler still running on stop");
+}
+
+Scheduler* Scheduler::GetScheduler() {
+  return current_scheduler;
+}
+
+void Scheduler::Set(Scheduler* scheduler) {
+  current_scheduler = scheduler;
+}
+
+Fiber* PollRandomElementFromList(std::vector<Fiber*>& list) {
+  auto rand_pos = GetRandNumber(list.size());
+  auto* next = list[rand_pos];
+  list.erase(list.begin() + rand_pos);
+  return next;
+}
+
+BiNode* PollRandomElementFromList(BiList& list) {
+  auto rand_pos = GetRandNumber(list.GetSize());
+  auto* next = list.GetNth(rand_pos);
+  list.Erase(next);
+  return next;
+}
+
+void Scheduler::Schedule(Fiber* fiber) {
+  _queue.push_back(fiber);
+  if (!IsRunning()) {
+    _running = true;
+    RunLoop();
+    _running = false;
+  }
+}
+
+Fiber* Scheduler::Current() {
+  return current;
+}
+
+Fiber::Id Scheduler::GetId() {
+  YACLIB_DEBUG(current == nullptr, "Current can't be null");
+  return current->GetId();
+}
+
+void Scheduler::TickTime() {
+  _time += 10;
```

Comment:
I think we can add ability to specify it

Reply:
Also it can be some computable parameter? something like max(min_value, min_time_difference);